dht.go

```diff
@@ -352,6 +359,44 @@ func makeDHT(ctx context.Context, h host.Host, cfg dhtcfg.Config) (*IpfsDHT, err
 	return dht, nil
 }
 
+// lookupCheck performs a lookup request to a remote peer.ID, verifying that it is able to
+// answer it correctly
+func (dht *IpfsDHT) lookupCheck(ctx context.Context, p peer.ID) error {
+	// lookup request to p requesting for its own peer.ID
+	peerids, err := dht.protoMessenger.GetClosestPeers(ctx, p, p)
+	// p should return at least its own peerid
+	if err == nil && len(peerids) == 0 {
+		return fmt.Errorf("peer %s failed to return its closest peers, got %d", p, len(peerids))
+	}
+	return err
+}
+
+// verifyAndLogPeerQuery returns true if p has been queried less than dht.lookupCheckInterval ago.
+// If the peer has not been queried recently, log that it is being queried.
+func (dht *IpfsDHT) verifyAndLogPeerQuery(p peer.ID) bool {
+	dht.recentlyCheckedPeersLk.Lock()
+	defer dht.recentlyCheckedPeersLk.Unlock()
+
+	now := time.Now()
+
+	// clean recentlyCheckedPeers
+	for peerid, t := range dht.recentlyCheckedPeers {
+		// remove peers that have been queried more than lookupCheckInterval ago
+		if t.Add(dht.lookupCheckInterval).Before(now) {
+			delete(dht.recentlyCheckedPeers, peerid)
+		}
+	}
+
+	// if p still in recentlyCheckedPeers, it has been queried less than
+	// lookupCheckInterval ago
+	_, ok := dht.recentlyCheckedPeers[p]
+	if !ok {
+		// log that the peer is being queried
+		dht.recentlyCheckedPeers[p] = now
```

Comment:
AFAIT This could grow infinitely if someone is able to trigger `verifyAndLogPeerQuery` which AFAIT would be easy to do with a sybil attacks and a bunch of peer ids that would land in unused buckets, the RT would then try to add the nodes in the buckets and add them here.
I would suggest adding a maximum size to the cache, if the map is full then try:
```go
for peerid, t := range dht.recentlyCheckedPeers {
		// remove peers that have been queried more than lookupCheckInterval ago
		if t.Add(dht.lookupCheckInterval).Before(now) {
			delete(dht.recentlyCheckedPeers, peerid)
		}
	}
```
And if that is still not enough remove random peers from the cache.